File: app/src/main/java/com/jacksonke/teresapassword/SiteEntity.java
package com.jacksonke.teresapassword;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;

/**
 * Created by jacksonke on 30/08/2018.
 *
 *
 */
@Entity
public class SiteEntity {
    @Id
    long id;

    String name;
    int ver;
    int type = Constant.TYPE_NUMBER_ABC;

    public SiteEntity(String siteName){
        id = 0;
        name = siteName;
        ver = 0;
        type = Constant.TYPE_NUMBER_ABC;
    }


    public SiteEntity(){
        id = 0;
        name = "";
        ver = 0;
    }

    @Override
    public String toString() {
        return name;
    }

    String getDebugString(){
        String ret = "";
        if (BuildConfig.DEBUG){
            ret = ret + " id:" + id
                + " name:" + name
                + " ver:" + ver
                + " type:" + type;
        }

        return ret;
    }
}

File: app/src/main/java/com/jacksonke/teresapassword/SiteEntity.kt
package com.jacksonke.teresapassword

import io.objectbox.android.BuildConfig
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id

@Entity
class SiteEntity {
    @Id
    var id: Long
    var name: String
    var ver: Int
    var type: Int = Constants.TYPE_NUMBER_ABC

    constructor(siteName: String) {
        id = 0
        name = siteName
        ver = 0
        type = Constants.TYPE_NUMBER_ABC
    }

    constructor() {
        id = 0
        name = ""
        ver = 0
    }

    override fun toString(): String {
        return name
    }

    val debugString: String
        get() {
            var ret = ""
            if (BuildConfig.DEBUG) {
                ret = (ret + " id:" + id
                        + " name:" + name
                        + " ver:" + ver
                        + " type:" + type)
            }
            return ret
        }
}